packages/terra-form-input/src/Input.jsx

```diff
@@ -122,7 +122,7 @@ class Input extends React.Component {
     const theme = this.context;
     const formInputClassNames = classNames(
       cx(
-        'form-input',
+        type === 'file' ? '' : 'form-input',
```

Comment:
Nit : To keep it consistent ( Unless changing like this throws lint error ).
```suggestion
       'form-input': (type !== 'file'),
```

Reply:
^ adding on to Supreeth's comment. The syntax below shows an example for conditional assignment using classNames.

```js
// These classes are only applied if the logic statement is truthy
{ 'form-error': isInvalid },
{ 'form-incomplete': (isIncomplete && required && !isInvalid) },
```

However, removing `form-input` will remove the majority of the [themable properties](https://github.com/cerner/terra-core/blob/50b6591cb451ee123e7e81d0ec082fe8151a5af1/packages/terra-form-input/src/Input.module.scss#L7-L64) of the component. This is probably not what we want to do, but we can track down the specific properties causing the styling issues to see if we can create a class to conditionally apply them.

Is there a specific set of styles causing the issue?

Reply:
+1 
keeping  class `form-input` as it was before and moving only specific styles causing truncation in IE to new conditional class looks more appropriate here.

Reply:
@neilpfeiffer to clarify our decision here so we can link back to it in future discussions. Is your recommendation to remove all terra styles and custom properties when the type is file? Specifically, does this also include removing the incomplete and error classes?

Reply:
Thanks @StephenEsser, adding details here for the confirmed decisions and future reference.

---

**Changes:**

Since specific input types have native UI provided by browsers by default, like `type="file"`, and only `type="text"`, `type="number"`, and `type="password"` were originally designed for and tested as the terra-form-input component with intentional styling and behavior handling, we've decided to create a separation for types that are 'supported' (has Terra specific styling and behaviors) and those that are 'unsupported'. 

Rather than completely blocking those types that are unsupported, the decision is to conditionally skip binding all terra classes: the base `'form-input'`, plus the conditional `'form-error'`, `'form-incomplete'`, and the `theme`, yet still pass through any consumer provided classNames. 

After testing, the final list of 'supported' and 'unsupported' `types` are as follows:

- **Supported Types** = 'text', 'number', 'password', 'email', 'search', 'tel', 'url', 'hidden'.

- **Unsupported Types** =  'date', 'datetime-local', 'month', 'time', 'week', 'color', 'file', 'range', 'checkbox', 'radio', 'button', 'reset', 'submit'.

---

Note: If future enhancements provide Terra styling and behavior handling for any of the unsupported types, they can be added back to the list of supported types.